portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -35,17 +36,18 @@
 public class DataServlet extends HttpServlet {
 
   private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final String COMMENT_TAG = "comment";
```

Comment:
As this appears to be a constant, should it be static? If so, java style would say to put constants at the top of your file generally.

Re the naming, is this a tag? Or better considered a key?

Re consistency, this string literal is extracted into a constant, but max_comments below isn't?

Reply:
Re: naming. I guess this should be considered a "kind". That is the official naming in the Firebase documentation. (Entities can have "kinds").

Re: consistency. I extracted this string literal because it was used in multiple places, but `max_comments` was not. Should I also extract `max_comments`?

Reply:
"kind" sounds good to me. The general test for "should X be a constant" should rarely be "in how many places is this used". When this is compiled, the code/memory code will be exactly the same whether you use the same string literal everywhere, or extract into a constant. I would ask, what is the most consistent, and thus the easiest for someone else reading my code to understand?

When reading through this, when I see one is extracted into a constant, and the other isn't, why first thought is, "this seems odd, am I missing some piece of information in reading this"? If I saw that you had used always a string literal, or always a constant, I would attribute it to stylistic choice, and not think too much of it. One of the highest priorities for all code should be to be as readable as possible - and that often requires thinking like someone who is completely new to a codebase, and may not have background on why you made choices you did.